### PR TITLE
Feature/cp 3415/call back logic with validation

### PR DESCRIPTION
**Pull request by nf2000**
Call back Logic

https://chillibean.atlassian.net/browse/CP-3415

### CODE REVIEW

File: nims-notes-things/src/Form.js [hunk 1]
```diff
@@ -22,57 +24,81 @@ const Form = (props) => {
     };
     setFormData(newForm);
   };
+  const handleOnSubmit = (event) => {
+    validateSubmit(event);
+  };
+
+  const validateSubmit = (event) => {
```
Comment (damonspencerroberts) [inline nims-notes-things/src/Form.js]: See my comment here for an approach to this issue: https://github.com/alexholmes99/notepadchallenge/pull/6#discussion_r813051361

File: nims-notes-things/src/Button.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import styled from "styled-components";
+
+const Button = (props) => {
+  return <ButtonStyle {...props}>{props.children}</ButtonStyle>;
+};
+
+const ButtonStyle = styled.button`
+  border-radius: 5000px;
```
Comment (damonspencerroberts) [inline nims-notes-things/src/Button.js]: why 5000 px for the border-radius?

File: nims-notes-things/src/Form.js [hunk 3]
```diff
@@ -22,57 +24,81 @@ const Form = (props) => {
     };
     setFormData(newForm);
   };
```
Comment (damonspencerroberts) [inline nims-notes-things/src/Form.js:27]: add spaces between functions

File: nims-notes-things/src/Form.js [hunk 4]
```diff
@@ -22,57 +24,81 @@ const Form = (props) => {
...
-    empty.map((data) => {
-      return <h2>key.data</h2>;
+  const handleOnBlur = (value, key) => {
+    setFormEmpty((previousState) => {
+      return {
+        ...previousState,
+        [key]: value === "",
+      };
     });
+    console.log(empty);
```
Comment (damonspencerroberts) [inline nims-notes-things/src/Form.js]: remove console

**Review by damonspencerroberts — CHANGES_REQUESTED** [review]
More change see my comment for validation on Alex's PR, link above

Also, see me tomorrow regarding file structure.

File: nims-notes-things/src/Button.js (hunk 2 above)
Comment (nf2000) [reply nims-notes-things/src/Button.js]: this was just a random number to make button edges rounded

File: nims-notes-things/src/Form.js [hunk 5]
```diff
@@ -23,56 +26,97 @@ const Form = (props) => {
...
+    let isValidated = true;
+    Object.keys(formData).forEach((data) => {
+      if (formData[data] === "") {
+        setFormEmpty((previousState) => {
+          return {
+            ...previousState,
+            [data]: true,
+          };
+        });
+        return (isValidated = false);
```
Comment (damonspencerroberts) [inline nims-notes-things/src/Form.js]: no need for brackets on the return statement here

File: nims-notes-things/src/Form.js [hunk 6]
```diff
@@ -23,56 +26,97 @@ const Form = (props) => {
...
+      } else {
+        return (isValidated = true);
```
Comment (damonspencerroberts) [inline nims-notes-things/src/Form.js]: no need for an else block

File: nims-notes-things/src/Form.js [hunk 7]
```diff
@@ -23,56 +26,97 @@ const Form = (props) => {
     setFormData(newForm);
   };
 
-  const handleOnBlur = (e, key) => {
-    empty.map((data) => {
-      return <h2>key.data</h2>;
+  const validateSubmit = (event) => {
```
Comment (damonspencerroberts) [inline nims-notes-things/src/Form.js]: no need for event argument here if you're not using it

File: nims-notes-things/src/Form.js [hunk 8]
```diff
@@ -23,56 +26,97 @@ const Form = (props) => {
...
+      } else {
+        return (isValidated = true);
+      }
+    });
+    return isValidated;
+  };
+
+  const handleOnSubmit = (event) => {
+    event.preventDefault();
+    const isValidated = validateSubmit(event);
```
Comment (damonspencerroberts) [inline nims-notes-things/src/Form.js]: no need to pass the event parameter in `validateSubmit`

**Review by damonspencerroberts — CHANGES_REQUESTED** [review]
Couple of small comments :)

File: nims-notes-things/src/Form.js [hunk 9]
```diff
@@ -23,56 +26,97 @@ const Form = (props) => {
     setFormData(newForm);
   };
 
-  const handleOnBlur = (e, key) => {
-    empty.map((data) => {
-      return <h2>key.data</h2>;
+  const validateSubmit = (event) => {
+    let isValidated = true;
+    Object.keys(formData).forEach((data) => {
```
Comment (damonspencerroberts) [inline nims-notes-things/src/Form.js]: rename `data` here to `key`

**Review by damonspencerroberts — APPROVED** [review]
Looks good :) Can merge this to main, and then update redux branch with the main branch...

**Pull request merged**